mux_test.go

```diff
@@ -1389,6 +1391,30 @@ func TestSubrouterErrorHandling(t *testing.T) {
 	}
 }
 
+// See: https://github.com/gorilla/mux/issues/200
+func TestPanicOnCapturingGroups(t *testing.T) {
```

Comment:
Just rewrite this function:

```
defer func() {
    if recover() == nil {
        t.Errorf("(%v) Expected panic, however test completed sucessfully.\n", test.title)
    }
}()
r.NewRoute().Path("/{type:(promo|special)}/{promoId}.json")
```

The rest of the test changes can then be removed. There are already tests for capturing groups.